src/include/txn.h

```diff
@@ -98,6 +98,7 @@ struct __wt_txn_state {
 	volatile uint64_t id;
 	volatile uint64_t pinned_id;
 	volatile uint64_t metadata_pinned;
+	volatile bool is_allocating;
```

Comment:
@michaelcahill, I added this variable in __wt_txn_state which is padded in cache line size for optimization.
I think it would trigger not a bit performance issue here because it would not extend the cache line size,
but I found that once the transaction id allocation is done, session->txn sets the flag WT_TXN_HAS_ID.
Do you think it is better to use that flag for indicating transaction is still allocating txn id instead of this additional boolean flag?

Reply:
I think it's better to have something in `WT_TXN_STATE` -- that is intended to be in a single cache line rather than having threads pull parts of `WT_TXN` into each CPU's cache.

I could go either way on whether to use a bit of `id` or a separate field.  The approach here (using a separate field) has the definite advantage that `id` always contains an ordinary transaction ID and no masking / special handling is required.

Reply:
Thank you @michaelcahill . Now I understand that using a boolean flag in `WT_TXN_STATE` would be better than using the flag `WT_TXN_HAS_ID`

Related to this one, it is still not clear for me to decide between using `is_allocating` or stealing a bit from `id`. The advantage if we steal one bit of `id` is that the code (below) in `__wt_get_snapshot` and `__txn_oldest_scan` would be simpler. But if you think the comment explanation is enough, I personally think it is good to separate the flag instead of stealing one bit of `id` (since this might be used later)

Reply:
Let's stick with a separate `bool`.